Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -150,4 +161,19 @@ void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DO
     workerGlobalScope.thread().workerLoaderProxy().postTaskToLoader(WTFMove(completionHandler));
 }
 
+std::optional<PermissionQuerySource> Permissions::sourceFromContext(const ScriptExecutionContext& context)
```

Comment:
Please mark function as static and move near the top of this cpp file.

Reply:
Fixed!